skimage/restoration/_guided_filter.py

```diff
@@ -0,0 +1,161 @@
+    radius : int
+        The half edge size of the window used to compute pixel statistics for
+        smoothing. The actual filter window size is 2*radius + 1.
+    guide : array-like, optional
+        Guide image to control the smoothing. If no guide is specified the
+        image will be used as its own guide. The guide image must have the same
+        number of rows and columns as the input image. If the input image has
+        multiple channels the guide image must either have one channel, or the
+        same number of channels as the input. If the guide has only one channel
+        each channel in the input is smoothed with the same guide, otherwise
+        each channel in the input is used to guide the output.
+
+    Returns
+    -------
+    filtered_image : ndarray
+        The filtered array, as a float.
+
+    References
+    ----------
+    .. [1] "Guided Image Filtering", by He et al. appearing in ECCV 2010.
+    http://research.microsoft.com/en-us/um/people/kahe/eccv10/
+
+    Notes
+    -----
+    Currently uses scipy uniform_filter, so is not O(n) as implemented in the
+    paper.
+
+    Examples
+    --------
+    >>> a = np.eye(3)
+    # If no guide is specified, the image itself is used.
+    >>> guided_filter(a, 0.1, 1)
+    array([[ 0.82811797,  0.10548081,  0.08231598],
+           [ 0.10548081,  0.75720884,  0.10548081],
+           [ 0.08231598,  0.10548081,  0.82811797]])
+    # An explicit guide image can be specified.
+    >>> guided_filter(a, 0.1, 1, guide=a*2)
+    array([[ 0.94438564,  0.03506486,  0.02870165],
+           [ 0.03506486,  0.91356598,  0.03506486],
+           [ 0.02870165,  0.03506486,  0.94438564]])
+    # A multichannel image can be guided with a single channel image.
+    >>> a = np.tile(a[..., np.newaxis], (1, 1, 3))
+    >>> b = a[:, :, 0]**2
+    >>> guided_filter(a, 0.1, 1, guide=b)
+    array([[[ 0.82811797,  0.82811797,  0.82811797],
+        [ 0.10548081,  0.10548081,  0.10548081],
+        [ 0.08231598,  0.08231598,  0.08231598]],
+       [[ 0.10548081,  0.10548081,  0.10548081],
+        [ 0.75720884,  0.75720884,  0.75720884],
+        [ 0.10548081,  0.10548081,  0.10548081]],
+       [[ 0.08231598,  0.08231598,  0.08231598],
+        [ 0.10548081,  0.10548081,  0.10548081],
+        [ 0.82811797,  0.82811797,  0.82811797]]])
+    # Or with a multi-channel image.
+    >>> b = a**2
+    >>> guided_filter(a, 0.1, 1, guide=b)
+    array([[[ 0.82811797,  0.82811797,  0.82811797],
+            [ 0.10548081,  0.10548081,  0.10548081],
+            [ 0.08231598,  0.08231598,  0.08231598]],
+           [[ 0.10548081,  0.10548081,  0.10548081],
+            [ 0.75720884,  0.75720884,  0.75720884],
+            [ 0.10548081,  0.10548081,  0.10548081]],
+           [[ 0.08231598,  0.08231598,  0.08231598],
+            [ 0.10548081,  0.10548081,  0.10548081],
+            [ 0.82811797,  0.82811797,  0.82811797]]])
+    """
+    if guide is not None:
```

Comment:
Minor. as None is default, I would flip the position of this block with the `else`block to start with `if guide is None`